CONTRIBUTING_TO_OPENSHIFT_PLUGINS.md

```diff
@@ -0,0 +1,426 @@
+
+* [OpenShift Client Plugin](https://github.com/jenkinsci/openshift-client-plugin)
+* [OpenShift Sync Plugin](https://github.com/jenkinsci/openshift-sync-plugin)
+* [OpenShift Login Plugin](https://github.com/jenkinsci/openshift-login-plugin)
+* [OpenShift Pipeline Plugin](https://github.com/jenkinsci/openshift-pipeline-plugin)  (*again, only in maintenance at this point*)
+
+And the wiki URLs are: 
+
+* [https://wiki.jenkins.io/display/JENKINS/OpenShift+Pipeline+Plugin](https://wiki.jenkins.io/display/JENKINS/OpenShift+Pipeline+Plugin)
+* [https://wiki.jenkins.io/display/JENKINS/OpenShift+Pipeline+Plugin](https://wiki.jenkins.io/display/JENKINS/OpenShift+Sync+Plugin)
+* [https://wiki.jenkins.io/display/JENKINS/OpenShift+Pipeline+Plugin](https://wiki.jenkins.io/display/JENKINS/OpenShift+Client+Plugin)
+* [https://wiki.jenkins.io/display/JENKINS/OpenShift+Pipeline+Plugin](https://wiki.jenkins.io/display/JENKINS/OpenShift+Login+Plugin)
+
+As the development process for each of these plugins are similar in many ways, we are documenting the process here under 
+the repository for the OpenShift Jenkins images, where each of the plugins will reference this document in their respective 
+READMEs.
+
+## Constructing your development sandbox
+
+You will need an environment with the following tools installed:
+
+* Maven (the `mvn` command)
+* Git
+* Java
+* an IDE or editor of your choosing
+
+The Jenkins open source project already has a bunch of useful links on setting up your development environment, including 
+launching Jenkins from the IDE (though there are pros and cons for developing this way vs. developing against a Jenkins 
+server running in an OpenShift pod).
+
+Here are a few of those links for reference:
+
+* [https://wiki.jenkins.io/display/JENKINS/Extend+Jenkins](https://wiki.jenkins.io/display/JENKINS/Extend+Jenkins) (note, there are many useful child pages under this wiki page)  
+* [https://wiki.jenkins.io/display/JENKINS/Plugin+tutorial](https://wiki.jenkins.io/display/JENKINS/Plugin+tutorial) (one of the key child pages under Extend Jenkins)
+* [https://wiki.jenkins.io/display/JENKINS/Setting+up+Eclipse+to+build+Jenkins](https://wiki.jenkins.io/display/JENKINS/Setting+up+Eclipse+to+build+Jenkins) (has concepts common to other IDEs undoubtedly)
+
+### Basic code development flow
+
+Our plugins are constructed such that if they are running in an OpenShift pod, they can determine how to connect to the 
+associated OpenShift master automatically, and no configuration of the plugin from the Jenkins console is needed.
+
+If you choose to run an external Jenkins server and you would like to test interaction with an OpenShift master, you will need to manually configure the plugin.  See each plugin's README or the OpenShift documentation
+for the specifics.
+
+An example flow when running in an OpenShift pod:
+
+1. Install maven (platform specific)
```

Comment:
Could probably skip this step as maven was specified as a tool that needs to be installed in the above section.

Reply:
ok